src/script/components/calling/CallingCell.tsx

```diff
@@ -448,163 +434,170 @@ const CallingCell: React.FC<CallingCellProps> = ({
+                        element?.focus();
+                      }
+                    }}
+                    className="call-ui__button call-ui__button--red call-ui__button--large"
+                    onClick={() => (isIncoming ? callActions.reject(call) : callActions.leave(call))}
+                    onBlur={() => clearShowAlert()}
+                    title={!isGroup && showAlert ? call1To1StartedAlert : t('videoCallOverlayHangUp')}
+                    aria-label={!isGroup && showAlert ? call1To1StartedAlert : t('videoCallOverlayHangUp')}
+                    type="button"
+                    data-uie-name="do-call-controls-call-decline"
+                  >
+                    <Icon.Hangup className="small-icon" style={{maxWidth: 17}} />
+                  </button>
+                </li>
+              )}
 
-              {isFullUi && (
-                <div
-                  className={cx('call-ui__participant-list__wrapper', {
-                    'call-ui__participant-list__wrapper--active': showParticipants,
-                  })}
-                >
-                  <FadingScrollbar className="call-ui__participant-list__container">
-                    <ul className="call-ui__participant-list" data-uie-name="list-call-ui-participants">
-                      {participants
-                        .slice()
-                        .sort((participantA, participantB) => sortUsersByPriority(participantA.user, participantB.user))
-                        .map((participant, index, participantsArray) => (
-                          <li key={participant.clientId} className="call-ui__participant-list__participant">
-                            <CallParticipantsListItem
-                              key={participant.clientId}
-                              callParticipant={participant}
-                              selfInTeam={selfUser && teamState.isInTeam(selfUser)}
-                              isSelfVerified={isSelfVerified}
-                              showContextMenu={!!isModerator}
-                              onContextMenu={event => getParticipantContext(event, participant)}
-                              isLast={participantsArray.length === index}
-                            />
-                          </li>
-                        ))}
-                    </ul>
-                  </FadingScrollbar>
-                </div>
+              {isIncoming && (
+                <li className="conversation-list-calling-cell-controls-item">
+                  {isDeclined ? (
+                    <button
+                      className="call-ui__button call-ui__button--green call-ui__button--join call-ui__button--join--large "
+                      onClick={answerCall}
+                      type="button"
+                      data-uie-name="do-call-controls-call-join"
+                    >
+                      {t('callJoin')}
+                    </button>
+                  ) : (
```

Comment:
`IsDeclined` can only be true at that stage if `ìsTemporaryUser` is also true.

In that case we show a larger join button and no decline button if a call is ongoing bit the temp user hasn't accepted it yet

Reply:
damn this logic is quite hard to follow 😅

Reply:
I could add a few comments, maybe rename
```
const isDeclined = !!reason && [CALL_REASON.STILL_ONGOING, CALL_REASON.ANSWERED_ELSEWHERE].includes(reason);
```
to `isNotAccepted`

Reply:
Nah, I think the problem lies mainly in the rendering part, things are a little too nested. But not sure there is an easy way to fix that now; :)